src/manage_sql_secinfo.c

```diff
@@ -3451,6 +3461,46 @@ update_scap_end ()
   return 0;
 }
 
+/**
+ * @brief Abort scap update.
+ */
+static void
+abort_scap_update ()
+{
+  g_debug ("%s: update timestamp", __func__);
+
+  if (sql_int ("SELECT EXISTS (SELECT schema_name FROM"
+               "               information_schema.schemata"
+               "               WHERE schema_name = 'scap');"))
+    {
+      update_scap_timestamp ();
+      sql ("UPDATE scap.meta SET value = "
+           "    (SELECT value from scap2.meta WHERE name = 'last_update')"
+           "  WHERE name = 'last_update';");
+      sql ("DROP SCHEMA scap2 CASCADE;");
+      /* View 'vulns' contains references into the SCAP schema, so it is
+       * removed by the CASCADE. */
+      create_view_vulns ();
+    }
+  else
+    {
+      /* reset scap2 schema */
+      manage_db_init ("scap");
+      manage_db_init_indexes ("scap");
+      manage_db_add_constraints ("scap");
```

Comment:
Is it a good idea to error check these calls ? In case, one of them fails, we may not proceed with the next ones. 
What do you think ?